Feature/2.3/cloudscale server

gaudenz opened the pull request:
##### ISSUE TYPE
 - New Module Pull Request

##### COMPONENT NAME
cloudscale_server

##### ANSIBLE VERSION
```
ansible 2.3.0 (feature/2.3/cloudscale_server e9acf74eb9) last updated 2017/01/12 13:37:43 (GMT +200)
  config file = /home/gaudenz/.ansible.cfg
  configured module search path = [u'/home/gaudenz/.ansible/plugins/library']
```

##### SUMMARY
This is a cloud module to create, start, stop and delete servers on the cloudscale.ch IaaS platform through their API.

gaudenz committed:
cloudscale: new module cloudscale_server

gaudenz committed:
cloudscale: integration test for cloudscale_server

gaudenz committed:
Add dots at the end of all sentences

gaudenz committed:
Use del dict[k] instead of dict.pop(k)

gaudenz committed:
Fail if state changes run into a timeout

If the state does not change until the timeout is over. Something
probably went wrong and it's safer to fail.

gaudenz committed:
Remove superflous empty line

resmo commented on test/integration/targets/cloudscale_server/meta/main.yml:
```diff
@@ -0,0 +1,2 @@
+dependencies:
+  - prepare_tests
```
AFAICS this role is missing, would you mind add it to the PR?

gaudenz replied:
@resmo I don't know why this role does not exist, but this dependency is listed in all test roles unter integration/targets so I thought this should be added by default.

resmo replied:
I see, I am not aware of the purpose but at least it does not hurt. Ok then

resmo reviewed (commented):
Over all a clean and straight forward module. Well done @gaudenz

gaudenz committed:
Check for allowed server state at module startup

For this moudle to be able to do anything the server must be in one of
the allowed states 'running', 'stopped' or 'absent'. All other states
are either error states or short lived transition states ('changeing').
In this case wait for the server to reach one of the allowed states or
fail after a timeout.

gaudenz committed:
Change timeout to 30 seconds

The timeout was always meant to be 30 seconds, but was set to 30 days
unintentionally.

gaudenz commented:
AFAICS the test failure has nothing to do with the changes in this PR. Is there a way to schedule a recheck?

resmo commented:
The only way you can do it is with force push IMHO

resmo commented:
I kicked the build again, all fine. 

shipit
